testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java

```diff
@@ -87,6 +87,31 @@ public void testArrayIndexingInView()
                 queryAssert -> queryAssert.containsOnly(row("hive")));
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testWithClause()
+    {
+        onHive().executeQuery("DROP VIEW IF EXISTS test_with_view");
+        onHive().executeQuery(
+                "CREATE VIEW test_with_view AS " +
+                        "WITH t AS (SELECT n_nationkey, n_regionkey FROM nation WHERE n_nationkey = 8) SELECT * FROM t");
+
+        assertThat(onHive().executeQuery("SELECT * FROM test_with_view"))
+                .containsOnly(row(8, 2));
+    }
```

Comment:
here & elsewhere:

- drop the view after the test (cleanup, but no need for try-finally)
- use `assertViewQuery` wherever possible, to assert we're returning same results as Hive does.